pages/x/index.tsx

```diff
@@ -17,18 +17,22 @@ const ThirdPartyRegistryList = () => {
     setQuery(event.target.value);
   }
 
-  const list = useMemo(
+  // Sorting the list only if entries are changed
+  const list = useMemo(() => {
+    return Object.keys(entries).sort((nameA, nameB) =>
+      nameA.localeCompare(nameB)
+    );
+  }, [entries]);
+
+  // Filtering the memoized list
+  const filteredList = useMemo(
     () =>
-      Object.keys(entries)
-        .filter(
-          (name) =>
-            name.toLowerCase().includes(query.toLowerCase()) ||
-            (entries[name].desc ?? "")
-              .toLowerCase()
-              .includes(query.toLowerCase())
-        )
-        .sort((nameA, nameB) => nameA.localeCompare(nameB)),
-    [entries, query]
+      list.filter((name) => {
+        const queryReg = new RegExp(query.split(" ").join(".*?"), "i");
```

Comment:
> The regex itself can be optimised: .* initially grabs as much as possible, then retracts only if the rest doesn't match. This takes a larger than neccessary amount of backtracking to find out. Instead, you could use a reluctant quantifier: .*?. This attempts to match as little as possible. If a match is found, it is found without backtracking. However, it still does a lot of backtracking if there is no match. We can do better: a[^b]*b (at this point, it doesn't really matter if the quantifier is greedy). A possessive quantifier (a[^b]*+b) would be even better, but javascript doesn't support these. Using a character class deprives us of a join, but see the next point.

For more details [stackexchange](https://codereview.stackexchange.com/a/23905)
For benchmark [fuzzy-match](https://jsperf.com/fuzzy-match)

> Note: Search will be faster

Reply:
Both greedy a[^b]*b and possessive a[^b]*+b regexes are not working mentioned in [stackexchange link](https://codereview.stackexchange.com/a/23905).

`
const queryWords = query.split(" ").filter((word) => word.length);
queryWords.reduce((a, b) => {
        if (a === "") return b;
        return a + "[^" + b + "]*" + b;
}, "")
`

Tried here:
https://regex101.com/r/spw1H6/1
https://regex101.com/r/ecgzzN/1

@rjoydip  Do you have any solution for supporting backtracking?

Reply:
@sandeshdamkondwar I don't have any solution right now. I will try to look solution later.

Reply:
Yes, will go that way. `a + ".*?" + b` is working fine. I'll revert back to that position. If you find any optimal solution, feel free to add in separate PR.